lib/purescript-cst/src/Language/PureScript/AST/Exported.hs

```diff
@@ -152,5 +166,13 @@ reorder refs =
   where
   refIndices =
     M.fromList $ zip (map declRefName refs) [(0::Int)..]
-  refIndex decl =
-    declName decl >>= flip M.lookup refIndices
+  refIndex = \case
+    KindDeclaration _ _ n _ ->
+      M.lookup (TyName n) refIndices <|> M.lookup (TyClassName (tyToClassName n)) refIndices
+    decl -> declName decl >>= flip M.lookup refIndices
+
+  -- Workaround to the fact that the kind's name's ProperNameType
+  -- isn't the same as the declaration's ProperNameType
+  -- when that declaration is a type class
+  tyToClassName :: ProperName 'TypeName -> ProperName 'ClassName
```

Comment:
Would you mind moving this into the `where` if it's only going to be used in `reorder`?

Reply:
Isn't it already part of that `where`? Or did you mean to move it into a `where` clause for just the `KindDeclaration` path? For example:
```haskell
KindDeclaration _ _ n _ ->
  -- code
  where
  tyToClasName :: ProperName 'TypeName -> ProperName 'ClassName
decl -> --
```

Reply:
Yes sorry, you're right, I misread and didn't notice the identation before `tyToClassName`. Never mind!

Reply:
In the latest updates, this is used twice now. So, I've moved it to a top-level member.